app/main.py

```diff
@@ -1,6 +1,19 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    former_functions = {}
+
+    def wrapper(*args, **kwargs) -> Any:
+        nonlocal former_functions
```

Comment:
this line redundan here

Reply:
can you please explain why the nonlocal line is rebundant? i thought we need to use it in order to permanently change the "former_functions" variable in the cache function's scope